tests/backtest/test_portfolio.py

```diff
@@ -1,30 +1,61 @@
 """Test the Portfolio class."""
 
+import pytest
+
 from coin_test.backtest import Portfolio
+from coin_test.util import Money, Ticker
 
 
 def test_portfolio(example_assets: dict) -> None:
     """Initialize correctly."""
-    example_cash = 10000.0
+    example_base_currency = Ticker("USDT")
 
-    p = Portfolio(example_cash, example_assets)
+    p = Portfolio(example_base_currency, example_assets)
 
-    assert p.cash == example_cash
+    assert p.base_currency == example_base_currency
     assert p.assets == example_assets
 
 
-def test_portfolio_no_constructor() -> None:
-    """Initialize correctly on an empty constructor."""
-    p = Portfolio()
+def test_no_base_currency(example_assets: dict) -> None:
+    """Fail if there is no base currency in the portfolio."""
+    example_base_currency = Ticker("DOGE")  # not in assets
 
-    assert p.cash == 0
-    assert p.assets == {}
+    with pytest.raises(ValueError):
+        Portfolio(example_base_currency, example_assets)
 
 
-def test_free_cash(example_assets: dict) -> None:
+def test_bad_assets() -> None:
+    """Fail if assets have the wrong tickers."""
+    example_base_currency = Ticker("USDT")
+    example_assets = {Ticker("USDT"): Money(Ticker("BTC"), 100)}
+
+    with pytest.raises(ValueError):
+        Portfolio(example_base_currency, example_assets)
+
+
+def test_reserved_assets(example_assets: dict) -> None:
     """Return free cash property."""
-    example_cash = 10000.0
+    example_base_currency = Ticker("USDT")
+
+    p = Portfolio(example_base_currency, example_assets)
+
+    assert Money(Ticker("USDT"), 0) == p.reserved[Ticker("USDT")]
+
+
+def test_free_cash(example_assets: dict) -> None:
+    """Initialize correctly."""
```

Comment:
This docstring seems wrong?

Reply:
also `free_cash` seems like a weird name, is there something like available that suits better